server/types/float32.go

```diff
@@ -51,12 +51,8 @@ func (b Float32Type) CollationCoercibility(ctx *sql.Context) (collation sql.Coll
 
 // Compare implements the DoltgresType interface.
 func (b Float32Type) Compare(v1 any, v2 any) (int, error) {
-	if v1 == nil && v2 == nil {
-		return 0, nil
-	} else if v1 != nil && v2 == nil {
-		return 1, nil
-	} else if v1 == nil && v2 != nil {
-		return -1, nil
+	if hasNulls, res := types.CompareNulls(v1, v2); hasNulls {
```

Comment:
`types.CompareNulls` is incorrect for all of these. That sorts null values according to MySQL rules, and we have to sort them according to Postgres rules.